tests/extension/oneapi_queue_empty/queue_empty_ext_test.cpp

```diff
@@ -0,0 +1,124 @@
+/*******************************************************************************
+//
+//  SYCL 2020 Conformance Test Suite
+//
+//  Copyright (c) 2023 The Khronos Group Inc.
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+//
+*******************************************************************************/
+#include "../../common/common.h"
+
+namespace queue_empty::tests {
+constexpr size_t buf_size = 10;
+constexpr size_t iter_num = 10000000;
+
+template <size_t iter_num, typename AccT>
+class Kernel {
+  AccT array_acc;
+
+ public:
+  Kernel(AccT acc) : array_acc(acc) {}
+  void operator()() const {
+    for (size_t i = 0; i < iter_num; i++) {
+      int val = sycl::sqrt(float(i));
+      array_acc[val % array_acc.size()] = i;
+    }
+  }
+};
+
+TEST_CASE(
+    "Test for \"Queue empty\" extension, check extension functionality"
+    "when no commands are submitted to the queue",
+    "[oneapi_queue_empty]") {
+#ifndef SYCL_EXT_ONEAPI_QUEUE_EMPTY
+  SKIP("SYCL_EXT_ONEAPI_QUEUE_EMPTY is not defined");
+#else
+  auto queue = sycl_cts::util::get_cts_object::queue();
+  auto is_empty = queue.ext_oneapi_empty();
+  CHECK(std::is_same_v<decltype(is_empty), bool>);
+  CHECK(is_empty);
+#endif
+}
+
+TEST_CASE(
+    "Test for \"Queue empty\" extension, check extension functionality"
+    "after commands are submitted to the queue, test 1",
+    "[oneapi_queue_empty]") {
+#ifndef SYCL_EXT_ONEAPI_QUEUE_EMPTY
+  SKIP("SYCL_EXT_ONEAPI_QUEUE_EMPTY is not defined");
+#else
+  auto queue = sycl_cts::util::get_cts_object::queue();
+
+  size_t array[buf_size];
```

Comment:
Do you need these arrays?

Reply:
I added these arrays to avoid compiler optimizations in kernel code.

Reply:
Which compiler optimization?
But my point is that would it work with just
```c++
sycl::buffer<size_t> array_buf(buf_size);
```
instead?

Reply:
Done.